ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -142,12 +143,16 @@
    * @param channel the {@link ReadableStreamChannel} containing the blob data.
    * @param futureResult the future that will contain the result of the operation.
    * @param callback the callback that is to be called when the operation completes.
+   * @param readyForPollCallback The callback to be used to notify the router of any state changes within this
+   *                             operation.
    * @param time the Time instance to use.
    * @throws RouterException if there is an error in constructing the PutOperation with the given parameters.
    */
   PutOperation(RouterConfig routerConfig, NonBlockingRouterMetrics routerMetrics, ClusterMap clusterMap,
       ResponseHandler responseHandler, BlobProperties blobProperties, byte[] userMetadata,
-      ReadableStreamChannel channel, FutureResult<String> futureResult, Callback<String> callback, Time time)
+      ReadableStreamChannel channel, FutureResult<String> futureResult, Callback<String> callback,
+      ReadyForPollCallback readyForPollCallback,
+      ByteBufferAsyncWritableChannel.ChannelEventListener channelEventListener, Time time)
```

Comment:
prefer naming this is as `chunkArrivalListener` since the `ReadableStreamChannel` passed in has the name `channel`.